management_api_app/service_bus/deployment_status_update.py

```diff
@@ -86,8 +86,8 @@ def update_status_in_database(workspace_repo: WorkspaceRepository, message: Depl
     result = False
 
     try:
-        workspace = workspace_repo.get_workspace_by_workspace_id(message.id)
-        workspace_repo.update_workspace(create_updated_deployment_document(workspace, message))
+        workspace = workspace_repo.get_resource_dict_by_id(message.id)
```

Comment:
It is not super ideal that we now have to return less specific items because we are making this very generic -- we are missing a lot of pydantic validation here and class.field access -- perhaps the WorkspaceRepository could still have a get_workspace_by_workspace_id that would call get_resource_dict_by_id and parse_obj_as Workspace on return?

Reply:
same here , because it can be workspace and workspace service (and user resource in the future) ... we want to retrieve it and update the deployment properties

Reply:
I had a suggestion for a solution, do you feel that this would not work?